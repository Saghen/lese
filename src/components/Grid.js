import styled, { css } from "styled-components";
import Base from "./Base";

import { getGridProperties } from "../helpers";

export default styled(Base)`
  ${getGridProperties}
`;
