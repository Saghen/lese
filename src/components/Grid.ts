import styled from "@emotion/styled";
import Base, { BaseProps } from "./Base";
import { propertyGenerator } from "../helpers";

export interface GridProps extends BaseProps {
  columns?: string;
  rows?: string;
  autoColumns?: string;
  autoRows?: string;
  columnGap?: string;
  rowGap?: string;
  gap?: string;
  xAlign?: boolean | string;
  yAlign?: boolean | string;
  align?: boolean | string;
}

export interface GridChildProps {
  columnSelf?: string;
  rowSelf?: string;
}

const getGridProperties = propertyGenerator<GridProps>([
  ["columns", { property: "grid-template-columns" }],
  ["rows", { property: "grid-template-rows" }],
  ["autoColumns", { property: "grid-auto-columns" }],
  ["autoRows", { property: "grid-auto-rows" }],
  ["columnGap", { property: "grid-column-gap" }],
  ["rowGap", { property: "grid-row-gap" }],
  ["gap", { property: "grid-gap" }],
  ["xAlign", { property: "justify-items", default: "center" }],
  ["yAlign", { property: "align-items", default: "center" }],
  ["align", { property: "place-items", default: "center center" }],
]);

export default styled(Base)<React.PropsWithChildren<GridProps>>`
  display: grid;
  ${getGridProperties}
`;
