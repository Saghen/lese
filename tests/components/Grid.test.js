import React from "react";
import renderer from "react-test-renderer";
import Grid from "../../src/components/Grid";
import { matchers } from "jest-emotion";

expect.extend(matchers);

test("grid layout props", () => {
  const treeLayout = renderer
    .create(
      <Grid
        columns="auto auto"
        rows="auto auto"
        columnGap="8px"
        rowGap="8px"
        gap="8px"
      >
        <div>Test</div>
      </Grid>
    )
    .toJSON();

  const treeAutoLayout = renderer
    .create(
      <Grid autoColumns="auto" autoRows="auto">
        <div>Test</div>
      </Grid>
    )
    .toJSON();

  const treeChildLayoutDefault = renderer
    .create(
      <Grid xAlign yAlign align>
        <div>Test</div>
      </Grid>
    )
    .toJSON();

  const treeChildLayout = renderer
    .create(
      <Grid xAlign="stretch" yAlign="end" align="normal start">
        <div>Test</div>
      </Grid>
    )
    .toJSON();

  expect(treeLayout).toMatchSnapshot();
  expect(treeAutoLayout).toMatchSnapshot();
  expect(treeChildLayoutDefault).toMatchSnapshot();
  expect(treeChildLayout).toMatchSnapshot();
});

test("grid child props", () => {
  const tree = renderer
    .create(
      <Grid columns="10px 10px" rows="10px 10px">
        <div column="2" row="2">
          Test
        </div>
      </Grid>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
