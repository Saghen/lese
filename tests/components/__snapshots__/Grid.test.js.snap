// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid child props 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: 10px 10px;
  grid-template-rows: 10px 10px;
}

.emotion-0 > *:nth-child(1) {
  grid-column: 2;
  grid-row: 2;
}

<div
  className="emotion-0"
>
  <div
    column="2"
    row="2"
  >
    Test
  </div>
</div>
`;

exports[`grid child props 2`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: 10px 10px;
  grid-template-rows: 10px 10px;
}

.emotion-0 > *:nth-child(1) {
  grid-column: 2;
  grid-row: 2;
}

.emotion-0 > *:nth-child(2) {
  grid-column: 2;
  grid-row: 2;
}

<div
  className="emotion-0"
>
  <div
    column="2"
    row="2"
  >
    Test
  </div>
  <div
    column="2"
    row="2"
  >
    Test
  </div>
</div>
`;

exports[`grid layout props 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: auto auto;
  grid-template-rows: auto auto;
  grid-column-gap: 8px;
  grid-row-gap: 8px;
  grid-gap: 8px;
}

<div
  className="emotion-0"
/>
`;

exports[`grid layout props 2`] = `
.emotion-0 {
  display: grid;
  grid-auto-columns: auto;
  grid-auto-rows: auto;
}

<div
  className="emotion-0"
/>
`;

exports[`grid layout props 3`] = `
.emotion-0 {
  display: grid;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  place-items: center center;
}

<div
  className="emotion-0"
>
  <div>
    Test
  </div>
</div>
`;

exports[`grid layout props 4`] = `
.emotion-0 {
  display: grid;
  justify-items: stretch;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  place-items: normal start;
}

<div
  className="emotion-0"
>
  <div>
    Test
  </div>
</div>
`;
